### DISCUSSION TITLE
add arguments to xcolor

### DISCUSSION THREAD

**Discussion by tpdsantos** [joaomlourenco/novathesis#439] [Q&A] [answered]
I was writing a documentation guide for some code that I made in _Julia_ and created a new syntax highlight with the `lstlisting` package:

```latex
\lstset{
  basicstyle=\ttfamily\small, 
  columns=fixed,
  numberstyle=\small\ttfamily\color{Gray},
  stepnumber=1,              
  numbersep=10pt, 
  numberfirstline=true, 
  numberblanklines=true, 
  tabsize=4,
  extendedchars=true,
  breaklines=true,        
  keywordstyle=\color{magenta},
  keywordstyle=[2]{\color{blue}},
  keywordstyle=[3]{\color{cyan}},
  identifierstyle=, % using emph or index keywords
  commentstyle=\sffamily\color{OliveGreen},
  stringstyle=\color{Maroon},
  showstringspaces=false,
  showtabs=false,
  upquote=false,
  texcl=true % interpet comments as LaTeX
}

\lstdefinelanguage{julia}{
  keywordsprefix=::,
  morekeywords = {exit,whos,edit,load,is,isa,isequal,typeof,tuple,ntuple,uid,hash,finalizer,convert,promote,subtype,typemin,typemax,realmin,realmax,sizeof,eps,promote_type,method_exists,applicable,invoke,dlopen,dlsym,system,error,throw,assert,new,Inf,Nan,pi,im,begin,while,for,in,return,break,continue,macro,quote,let,if,elseif,else,try,catch,end,bitstype,ccall,do,using,module,import,export,importall,baremodule,immutable,local,global,const,Bool,Integer,Int8,Int16,Int32,Int64,Uint,Uint8,Uint16,Uint32,Uint64,Real,Float,Float32,Float64,Complex64,Complex128,Any,Nothing,AbstractString,String,Char,None,function,type,typealias,abstract,ispath},
  morekeywords=[2]{@info,@eval,@pump,@valve,@valves,@step,@uvc,@monitor,@csinfo,@livalve,sleep,OceanOptics,RS232,K501_Pump,S1050_Pump,open_port,write,str_from_serialport,UVcell,SartoriusBalance,calibrate_pump,monitor_balance,update_calibs,reset_calibs,rm_calibs,ValveType,pulse_experiment,equilibration},
  morekeywords=[3]{true,false},
  sensitive=true,
  morecomment=[l]{\#},
  morestring=[b]',
  morestring=[b]" 
}
```

I was trying to add this code to the template but an error occurred stating the `OliveGreen` color was not defined. To add this color name I added in the preamble `\usepackage[usenames,dvipsnames]{xcolor}`, which is what I had to add in my original file. This led to the error that the `xcolor` package had already been called with different arguments. However I couldn't find the explicit call to this package, which leads me to believe it is called internally in the `memoir` class. Is there a way to safely add these extra arguments to the `xcolor` package?

Thanks in advance

**Comment by joaomlourenco** [comment, marked as answer]
In file `Config/0_memoir.tex` add
```latex
\PassOptionsToPackage{usenames,dvipsnames}{xcolor}
```
and let me know if it worked as you expected.

**Reply by tpdsantos** [reply]
Thank you for the reply. I added that line in the file and it did not work... I also double checked if those options are the actual solution for the problem and they are. Is there any other way?

**Reply by joaomlourenco** [reply]
Why do you say it did not work?  An error?  Options silently ignored?  Something else?

**Reply by joaomlourenco** [reply]
I confirm that with
```latex
\PassOptionsToPackage{usenames,dvipsnames}{xcolor}
``` 
in file `Config/0_memoir.tex add` the color `OliveGreen` is properly recognized.

**Reply by tpdsantos** [reply]
You're right, I didn't specify. In my case it was silently ignored, adding `\PassOptionsToPackage{usenames,dvipsnames}{xcolor}` did not make the color `OliveGreen` to be recognized. With a bit of research I found that for it to work it has to be passed before the package is called. I looked into `novathesis.cls` and the `0_memoir.tex` file is called after the `xcolor` package is called. I'm using the `6.10.1` version

**Reply by tpdsantos** [reply]
I downloaded the `7.0.0` version and `0_memoir.tex` is indeed called before package loading, that's why I think it works for you but not for me. Anyway, I added the extra argument directly in `novathesis.cls` and it all works as expected.

**Reply by tpdsantos** [reply]
I'll mark this one as the correct answer because it works for the latest version released.

**Reply by joaomlourenco** [reply]
That's why it is really important to properly identify the working version.
I **strongly** suggest you to switch to the latest version.  It is 99% compatible (the name of some options changed in `Config/1_novathesis.tex`.  All the rest shall be the same.

**Reply by tpdsantos** [reply]
I totally agree, my fault here, I didn't realize there was a newer version, will do !
